File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/lock/scanner/ScanForLockUseCase.java
package com.lattis.ellipse.domain.interactor.lock.scanner;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.lock.base.BaseLockUseCase;
import com.lattis.ellipse.domain.model.ScannedLock;
import com.lattis.ellipse.domain.repository.BluetoothRepository;
import com.lattis.ellipse.domain.repository.LockRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 4/26/17.
 */

public class ScanForLockUseCase extends BaseLockUseCase<ScannedLock> {

    @Inject
    ScanForLockUseCase(ThreadExecutor threadExecutor,
                       PostExecutionThread postExecutionThread,
                       BluetoothRepository bluetoothRepository,
                       LockRepository lockRepository
                       ) {
        super(threadExecutor, postExecutionThread,bluetoothRepository,lockRepository);
    }

    @Override
    protected Observable<ScannedLock> buildUseCaseObservable() {
        return getBluetoothRepository().startScan(35000);
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/lock/scanner/ScanForLockUseCase.kt
package com.lattis.domain.usecase.lock.scanner

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.BluetoothRepository
import com.lattis.domain.repository.LockRepository
import com.lattis.domain.usecase.lock.base.BaseLockUseCase
import com.lattis.domain.models.ScannedLock
import com.lattis.domain.repository.SaSOrPSLockRepository
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class ScanForLockUseCase @Inject internal constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    bluetoothRepository: BluetoothRepository,
    lockRepository: LockRepository,
    saSOrPSLockRepository: SaSOrPSLockRepository
) : BaseLockUseCase<ScannedLock>(
    threadExecutor,
    postExecutionThread,
    bluetoothRepository,
    lockRepository,
    saSOrPSLockRepository
) {
    override fun buildUseCaseObservable(): Observable<ScannedLock> {
        return bluetoothRepository.startScan(35000)
    }
}